ta/Makefile

```diff
@@ -44,8 +54,10 @@ all: ta
 
 .PHONY: ta
 ta: 
-	$(q)$(foreach dir,$(TA_DIRS), $(MAKE) -C $(dir) O=$(O)/$(dir) &&) true
+	$(q)$(foreach dir,$(TA_DIRS), $(MAKE) -C $(dir) O=$(out-dir)/ta/$(dir) &&) true
 
 .PHONY: clean
 clean:
-	$(q)$(foreach dir,$(TA_DIRS), $(MAKE) -C $(dir) O=$(O)/$(dir) $@ &&) true
+	$(q)$(foreach dir,$(TA_DIRS), $(MAKE) -C $(dir) O=$(out-dir)/ta/$(dir) $@ &&) true
+	$(q)$(foreach dir,$(TA_DIRS), bash -c 'cd $(dir) && rmdir $(out-dir)/ta 2> /dev/null';) true
+	$(q)$(foreach dir,$(TA_DIRS), bash -c 'cd $(dir) && rmdir $(out-dir) 2> /dev/null';) true
```

Comment:
Just make sure that `$(SHELL)` is initialized properly, everyone doesn't use bash as their normal shell.